### ISSUE TITLE
Schematize the documentation

### ISSUE THREAD

**Issue by fvisin** [mila-udem/blocks#831]
I would like to know your opinion on 2 different issue I see with the documentation:

1. I think the introduction tutorial is too broad and can be confusing as it introduces too many concepts without completely defining some of them. It is a very good *Overview* of Blocks, but as an *Introduction* I think it would be better to stick to a very basic example of the core of Blocks, e.g., the same example without graph management, regularization and monitoring.

2. It might be of use for experienced Blocks users to have access to a "schematized" version of the documentation.

Let me elaborate on the second issue. I recently went through the documentation of Blocks for the n-th time. I know how Blocks works, but since I haven't been using it for a while I wanted to refresh my memory and check a couple of things I was not sure about. I ended up reading it all again for the (n+1)th time.

I think the available documentation is great for new users as it goes through every step adding knowledge one step at the time, but for people who want to have a quick access to specific information it might be too verbose.

I propose to add at the end of each Tutorial a "Tip box" or a final section, to wrap up the content of the tutorial. Here is an example of what I would do for the Bricks tutorial:

## Bricks:
* Bricks take Theano variables as inputs, and provide Theano variables as outputs.
* Bricks are defined by:
    * **a set of attributes:** specifying the configuration of the Block (e.g., the number of input and output units)
    * **a set of parameters:** representing the parameters of the brick object that will vary during learning (e.g., the weights and the biases).

### Life cycle of a Brick:

* **Configuration:** set (part of) the **attributes** of the brick. 
    * When the brick is created, by setting the arguments of the constructor
	* Later, by setting the attributes of the brick object. 
	* No Theano variable is created in this phase.
* **Allocation (optional):**  allocate the **Theano shared variables** for the parameters of the Brick. 
    * When `Brick.allocate()` is called, the required Theano variables are allocated and initialized by default to `NaN`.
    * If not done explicitly, Blocks will quietly call `allocate()` when `apply()` is called.
* **Application:** instantiate a part of the **Theano computational graph**. 
    * Links the inputs and the outputs of the brick through its parameters and according to the attributes. 
    * Results in an error if the Brick object is not fully configured.
* **Initialization: **set the **numerical values** of the Theano variables that store the parameters of the Brick.

### Lazy initialization:
* You can create a Brick without *configuring* it fully
* This allows you to configure multiple Bricks together before applying (building) them

### Nested bricks:
* Bricks can have *children!*
* A parent Brick will configure its children automatically for you when you call its `apply()` method
* A parent Brick will initialize its children for you when you call its `initialize()` method

**Comment by dmitriy-serdyuk** [comment]
Yes, it make sense, a good summary always helps to study new stuff. 

I can also suggest to add links to blocks parts used in a tutorial to a 'See also' section at the end. So a person can easily study new bricks, algorithms, etc if he wants to.

**Comment by fvisin** [comment]
See https://github.com/mila-udem/fuel/pull/225/#discussion_r39677979 for inspiration on schematic summaries and https://github.com/mila-udem/fuel/pull/225/#issuecomment-141172089 for a suggestion on a cheat sheet that merges all the summaries together.
